### DISCUSSION TITLE
hi i am new in serenity

### DISCUSSION THREAD

**Discussion by amanjain19071975** [serenity-is/Serenity#6755] [Help]
how to use getLookupAsync on row level

**Comment by heleopless50** [comment]
i think this may help you refresh the data after saving
-adding [LookupScript("Products.Products",Expiration = -1)] to the product row 
-using Q.reloadLookup(ProductsRow.lookupkey) on the ProductDialog like this 
        protected onSaveSuccess(response: Q.SaveResponse): void {
            Q.reloadLookup(Products.ProductsRow.lookupKey);
        }

**Comment by amanjain19071975** [comment]
thanks for answer
but i need to use getlookupsync

**Reply by reach2rv** [reply]
Use service lookup if you do not want to refresh page and get updated data.

**Comment by heleopless50** [comment]
ou should note that the Q.getLookup method is obsolete, and you should prefer the Q.getLookupAsync instead. We merely use it here for sampling.

The first one does a synchronous XHR request, blocking the main browser UI thread, while the second one works asynchronously and returns a promise.

If you use Q.getLookup you'll be blocking the browser tab while the request is in progress, and will get a warning in the browser console like the one below:
